src/ManagedArray.hpp

```diff
@@ -156,6 +156,19 @@ class ManagedArray {
    */
   CHAI_HOST_DEVICE T& operator[](const int i) const;
 
+  CHAI_HOST_DEVICE ManagedArray(T* data, ExecutionSpace space, uint m_elems, bool owned);
```

Comment:
I don't think this needs to support pointers that came from an existing ManagedArray.